frontend/src/App.js

```diff
@@ -26,91 +26,93 @@ import ListView from './components/ListView';
-                    /> :
-                    <ListView /> }
-                <ToggleView view={this.props.view} changeView={this.props.changeView}/>
-            </div>
-        );
-    }
+  constructor(props) {
+    super(props);
+
+
+    // TODO: fetch /census_division_aggregate
+
+    firebase.auth().onAuthStateChanged(user => {
+      if (user) {
+        // User is signed in.
+        firebase.auth().currentUser.getIdToken(true)
+          .then(idToken => {
+            user.token = idToken;
+            user.db = firebase;
+            // Save user to state
+            this.props.signIn(user);
+            this.props.getServiceProviders(idToken);
+          }).catch(err => {
+            console.log(err);
+          });
+      } else {
+        // No user is signed in
+        this.props.getServiceProviders();
+      }
+    });
+
+  }
+  render() {
+    return (
+      <div>
+        <Sidebar
+          censusDivision={this.props.censusDivision}
+          demographic={this.props.demographic}
+          changeSelectedCensusDivision={this.props.changeSelectedCensusDivision}
+          changeDemographic={this.props.changeDemographic}
+          operatingBudget={this.props.operatingBudget}
+          clientServed={this.props.clientServed}
+          staffCount={this.props.staffCount}
+          changeOperatingBudget={this.props.changeOperatingBudget}
+          changeClientServed={this.props.changeClientServed}
+          changeStaffCount={this.props.changeStaffCount}
+        />
+          {this.props.view === constants.MAP_VIEW ?
+          <Map
+            selectedCensusDivision={this.props.selectedCensusDivision}
+            changeSelectedCensusDivision={this.props.changeSelectedCensusDivision}
+            /> :
+          <ListView />}
+        <ToggleView view={this.props.view} changeView={this.props.changeView}/>
+      </div>
+    );
+  }
 }
 
 function mapStateToProps(state) {
   return {
-    view: state.changeViewReducer.view,
-    censusDivision: state.changeDropdownReducer.censusDivision,
-    demographic: state.changeDropdownReducer.demographic,
-    operatingBudget: state.changeSliderReducer.operatingBudget,
-    clientServed: state.changeSliderReducer.clientServed,
-    staffCount: state.changeSliderReducer.staffCount,
-    serviceProviders: state.serviceProviderReducer,
-    user: state.authReducer,
+    view:                   state.changeViewReducer.view,
+    censusDivision:         state.changeDropdownReducer.censusDivision,
+    demographic:            state.changeDropdownReducer.demographic,
+    operatingBudget:        state.changeSliderReducer.operatingBudget,
+    clientServed:           state.changeSliderReducer.clientServed,
+    staffCount:             state.changeSliderReducer.staffCount,
+    user:                   state.authReducer,
     selectedCensusDivision: state.selectCensusDivisionReducer.selectedCensusDivision,
   };
 }
 
 function mapDispatchToProps(dispatch) {
   return {
```

Comment:
From the redux source code, ([See here](https://github.com/reduxjs/redux/blob/master/src/bindActionCreators.js)), it seems like
```
return bindActionCreators({
  changeView,
  changeCensusDivision,
  ...
}, dispatch)
```
Is equivalent and a bit cleaner b/c we don't have to write several `bindActionCreators`.
Is there a difference between that and the arrow function dispatchers like `signIn`?